code-challenges.js

```diff
@@ -24,11 +53,47 @@ var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
 
 
+// Writing off this function that will try its best, it will try everything it can. It wont worry what arguements get passed to it when the console.log is away. 
+const jimmyEatWorld = (string) => {
+   let theRide = string.substr(string.length - 1 >>> 1, (~string.length & 1) + 1);
+    return theRide
+}
+
+//Another full disclosure, I have been trying to play with substr since the last assessments and honestly its still kind of wizard space magic. Had to google this and I found the answer above, but its still super fuzzy to me. I'd love to have someone explain to me how substr works a little better.  Should have asked on the retro :(
+//Calling the functions below. 
+console.log(jimmyEatWorld(middleLetters1))
+console.log(jimmyEatWorld(middleLetters2))
+
 
 
 // --------------------4) Create a class to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
 
 
+//Quick shoutout, once again, to Cruz for helping me understand this question a bit. He also wanted me to make everything a TACO but I refused. 
+
+//Creating the class Sphere that will have its value set to 0. 
+class Sphere{
+    constructor(){
+        this.sphere = 0;
+    }
+// Method to take in a sphere size argument and spit out the cool calculations. 
+    makeSphere(nums){
+        this.sphere = Math.floor(((nums**2)*4)*Math.PI);
+        //This will log out the area when the function is called.
+        console.log("Area of sphere is: ", this.sphere);
```

Comment:
all functions need a return in order to officially end the function. console.log can be used as a dev tool but try and return values instead of logging them :)